core/src/main/scala/shapeless/labelled.scala

```diff
@@ -79,7 +79,9 @@ trait FieldPoly extends Poly1 {
 trait FieldOf[V] {
   import labelled._
 
-  def ->>(v: V): FieldType[this.type, V] = field[this.type](v)
+  type FieldType = labelled.FieldType[this.type, V]
+
+  def ->>(v: V): labelled.FieldType[this.type, V] = field[this.type](v)
```

Comment:
If I try to specify `FieldType` as the return type, `v` is required to be `Nothing`. I don't know if we can fix this ?

```
[error] /home/peter/programming/shapeless/core/src/main/scala/shapeless/labelled.scala:85: type mismatch;
[error]  found   : v.type (with underlying type V)
[error]  required: Nothing
[error]   def ->>(v: V): FieldType = (field[this.type](v): FieldType)
```